diff --git a/src/server/game/Spells/Auras/SpellAuraEffects.cpp b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
index eb470bd9e..f66e072f3 100644
--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
@@ -5588,6 +5603,15 @@ void AuraEffect::HandleAuraDummy(AuraApplication const* aurApp, uint8 mode, bool
                     case 30105:
                         finalSpelId = 30104;
                         break;
+                    case 80866:
+                        finalSpelId = 80869;
+                        break;
+                    case 80867:
+                        finalSpelId = 80870;
+                        break;
+                    case 80868:
+                        finalSpelId = 80871;
+                        break;
                 }
 
                 if (finalSpelId)
